packages/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart

```diff
@@ -220,7 +220,7 @@ class CommitLogKeyStore extends BaseCommitLogKeyStore {
 
   bool _isSpecialKey(String atKey) {
     return atKey.contains(AtConstants.atEncryptionSharedKey) ||
```

Comment:
this `contains AtConstants.atEncryptionSharedKey` check is also a bug. Why are we treating keys that contain "shared_key" as special?

And if we are treating them as special for good reason, we should specifically only match `shared_key\..*@this_atservers_at_sign` or `@some_other_atsign:shared_key@this_servers_at_sign` (precise regexes are in at_commons in `Regexes._reservedKeysWithAtsignSuffix`) ... currently this code would also match keys like "@bob:key1.shared_keys.buzz@alice"

Reply:
@murali-shris I don't see a good explanation for why we are not treating the two reserved shared_key variants as 'special' which (as I understand it) would result in those keys not being synced?

Reply:
Oh - ignore that last comment. I understand it now. This ensures that those keys **_are_** synced, regardless of whether they match the enrollment namespace or not.